pkg/controllers/managementuser/clusterauthtoken/cluster_auth_token.go

```diff
@@ -0,0 +1,87 @@
+package clusterauthtoken
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	mgmtcontrollers "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	clusterv3 "github.com/rancher/rancher/pkg/generated/norman/cluster.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type clusterAuthTokenHandler struct {
+	tokenCache  mgmtcontrollers.TokenCache
+	tokenClient mgmtcontrollers.TokenClient
+}
+
+// Sync ClusterAuthToken back to Token.
+func (h *clusterAuthTokenHandler) sync(key string, clusterAuthToken *clusterv3.ClusterAuthToken) (runtime.Object, error) {
+	if clusterAuthToken == nil || clusterAuthToken.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if !clusterAuthToken.Enabled ||
+		isExpired(clusterAuthToken) ||
+		clusterAuthToken.LastUsedAt == nil {
+		return clusterAuthToken, nil // Nothing to do.
+	}
+
+	tokenName := clusterAuthToken.Name
+	token, err := h.tokenCache.Get(tokenName)
+	if err != nil {
+		if apierrors.IsNotFound(err) || apierrors.IsGone(err) {
+			return clusterAuthToken, nil // ClusterAuthToken was orphaned.
```

Comment:
Not necessarily in this PR but we may consider proactively deleting the orphaned ClusterAuthToken here as we do with orphaned ClusterUserAttributes. 🤔